### PR TITLE
2.14 release notes

### PR DESCRIPTION
**Pull request by narqo**
#### What this PR does

~~This is a draft PR that I'll populate with the 2.14 release notes.~~

These are the release notes for the 2.14 release.

#### Which issue(s) this PR fixes or relates to

Relates to #9382

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: docs/sources/mimir/release-notes/v2.14.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+- the `-querier.prefer-streaming-chunks-from-store-gateways` option in querier and store-gateway
+- the `-query-scheduler.use-multi-algorithm-query-queue` option in querier-scheduler
+- the YAML configuration `frontend.align_queries_with_step` in query-frontend
+
+## Experimental features
+
+Grafana Mimir 2.14 includes new features that are considered experimental and disabled by default.
+Use them with caution and report any issues you encounter:
+
+A new experimental Kafka-based ingest storage architecture is added, that decouples the write and read path through a Kafka-compatible backend.
```
Comment (56quarters) [inline docs/sources/mimir/release-notes/v2.14.md:84]: I'm not sure we want to mention this in release notes since it doesn't have any documentation yet or a migration path.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+- the `-ingester.return-only-grpc-errors` option in ingester
+- the `-ingester.client.circuit-breaker.*` options in ingester
+- the `-ingester.limit-inflight-requests-using-grpc-method-limiter` option in ingester
+- the `-ingester.client.report-grpc-codes-in-instrumentation-label-enabled` option in distributor and ruler
+- the `-distributor.limit-inflight-requests-using-grpc-method-limiter` option in distributor
+- the `-distributor.enable-otlp-metadata-storage` option in distributor
+- the `-ruler.drain-notification-queue-on-shutdown` option in ruler
+- the `-querier.max-query-into-future` option in querier
+- the `-querier.prefer-streaming-chunks-from-store-gateways` option in querier and store-gateway
+- the `-query-scheduler.use-multi-algorithm-query-queue` option in querier-scheduler
```
Comment (chencs) [inline docs/sources/mimir/release-notes/v2.14.md:76]: Do we make note of _all_ configuration options removed, or just the stable/non-experimental ones? This one was experimental.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+<!-- Release notes are often future focused -->
+
+Grafana Labs is excited to announce version 2.14 of Grafana Mimir.
+
+The highlights that follow include the top features, enhancements, and bug fixes in this release.
+For the complete list of changes, refer to the [CHANGELOG](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
+
+## Features and enhancements
+
+The minimal supported version of Go is updated to 1.22.
```
Comment (charleskorn) [inline docs/sources/mimir/release-notes/v2.14.md:21]: Is this relevant to users? They just use the binary we build, right?

File: docs/sources/mimir/release-notes/v2.14.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+Use them with caution and report any issues you encounter:
+
+A new experimental Kafka-based ingest storage architecture is added, that decouples the write and read path through a Kafka-compatible backend.
+When enabled, distributors write incoming write requests to the Kafka-compatible backend, and the ingesters asynchronously replay ingested data from Kafka.
+
+New related configuration options that enable and configure the new architecture in Mimir are: `-ingest-storage.enabled`, `-ingest-storage.kafka.*`,
+`-ingest-storage.ingestion-partition-tenant-shard-size`, `-ingest-storage.read-consistency`, `-ingest-storage.migration.distributor-send-to-ingesters-enabled`,
+`-ingester.partition-ring.*`.
+
+Querier has an experimental streaming PromQL engine. The configuration option `-querier.query-engine=mimir` enables the engine.
```
Comment (charleskorn) [inline docs/sources/mimir/release-notes/v2.14.md:91]: I wouldn't mention this yet, I wouldn't want to encourage people to start using it just yet.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+Ingester added experimental `-ingester.ignore-ooo-exemplars` configuration. When set, the out of order exemplars are no longer reported
+to the remote write client.
+
+Querier now supports the experimental `limitk()` and `limit_ratio()` PromQL functions. The feature is disabled by default,
+but can be enabled with the `-querier.promql-experimental-functions-enabled=true` setting in the query-frontend and querier.
+
+Querier supports for the experimental `X-Mimir-Chunk-Info-Logger` header, that triggers logging information about TSDB chunks loaded from ingesters and store-gateways in the querier.
+The header should contain the comma separated list of labels, for which their value will be included in the logs.
```
Comment (charleskorn) [inline docs/sources/mimir/release-notes/v2.14.md:100]: I'm not sure this is worth mentioning, this is a feature intended to aid with debugging incorrect query results.

File: docs/sources/mimir/release-notes/v2.14.md (hunk 3 above)
Comment (pstibrany) [reply docs/sources/mimir/release-notes/v2.14.md:21]: As Charles says, this is not relevant. Furthermore mimir binaries are built with Go 1.23.1.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+Compactor adds a new `cortex_compactor_disk_out_of_space_errors_total` counter metric, that tracks how many times a compaction failed
+due to the compactor being out of disk. A new related alert `MimirCompactorHasRunOutOfDiskSpace` and it's runbook are added.
+
+Incoming OTLP requests were previously size-limited with the distributor's `-distributor.max-recv-msg-size` configuration.
+Distributor has a new `-distributor.max-otlp-request-size` configuration for limiting OTLP requests. The default value is set to limit request to 100 MiB.
+
+Distributor now replies with `Retry-After` header on retryable errors. The feature is controlled by the `-distributor.retry-after-header.enabled` configuration,
+that is enabled by default.
+
```
Comment (pstibrany) [inline docs/sources/mimir/release-notes/v2.14.md:40]: One new feature that we could mention is support for marking ingesters as read-only, that is used in updated downscaling procedure (PR to be merged: https://github.com/grafana/mimir/pull/9208)

File: docs/sources/mimir/release-notes/v2.14.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+Alertmanager adds the `-alertmanager.utf8-migration-logging-enabled` configuration option, that enables logging of tenant configurations,
+incompatible with UTF-8 strict mode. The default value for the configuration option is `false`.
+
+Compactor adds a new `cortex_compactor_disk_out_of_space_errors_total` counter metric, that tracks how many times a compaction failed
+due to the compactor being out of disk. A new related alert `MimirCompactorHasRunOutOfDiskSpace` and it's runbook are added.
+
+Incoming OTLP requests were previously size-limited with the distributor's `-distributor.max-recv-msg-size` configuration.
+Distributor has a new `-distributor.max-otlp-request-size` configuration for limiting OTLP requests. The default value is set to limit request to 100 MiB.
+
+Distributor now replies with `Retry-After` header on retryable errors. The feature is controlled by the `-distributor.retry-after-header.enabled` configuration,
```
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply docs/sources/mimir/release-notes/v2.14.md:38]: i think we now just enable it by default. Maybe we can mention what this means for users? It's that recovering from a Mimir outage should be easier now as the prometheus instances should respect Retry-After. ideally we can also mention how they should configure prometheus to respect retry-after headers. I'm not sure what the impact is for OTEL users

File: docs/sources/mimir/release-notes/v2.14.md (hunk 4 above)
Comment (56quarters) [inline docs/sources/mimir/release-notes/v2.14.md:91]: This should probably be the highlight of the release and deserves a longer description

File: docs/sources/mimir/release-notes/v2.14.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+The minimal supported version of Go is updated to 1.22.
+
+The streaming of the chunks from store-gateways to queriers is enabled by default.
+
+Alertmanager adds the `-alertmanager.log-parsing-label-matchers` configuration option to control logging when parsing label matchers.
+This flag is intended to be used with `-alertmanager.utf8-strict-mode-enabled` to validate UTF-8 strict mode is working as intended.
+The default value for the configuration option is `false`.
+
+Alertmanager adds the `-alertmanager.utf8-migration-logging-enabled` configuration option, that enables logging of tenant configurations,
```
Comment (56quarters) [inline docs/sources/mimir/release-notes/v2.14.md:29]: The alertmanager flags, compactor alert, and distributor changes don't seem like they're worth calling out in release notes.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+Distributor now replies with `Retry-After` header on retryable errors. The feature is controlled by the `-distributor.retry-after-header.enabled` configuration,
+that is enabled by default.
+
+## Important changes
+
+In Grafana Mimir 2.14 the following behavior has changed:
+
+When executing a remote read request, the querier now honors the start/end time range, specified in the read hints.
+
+To prevent queue starvation with querier-worker queue prioritization algorithm, the querier now sets the minimum `-querier.max-concurrent` to four.
```
Comment (56quarters) [inline docs/sources/mimir/release-notes/v2.14.md:47]: This seems pretty obscure

File: docs/sources/mimir/release-notes/v2.14.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+Distributor configuration's `-distributor.retry-after-header.max-backoff-exponent` and `-distributor.retry-after-header.base-seconds` are replaced
+with `-distributor.retry-after-header.min-backoff` and `-distributor.retry-after-header.max-backoff` for easier configuration.
+
+The circuit breakers in ingester are improved to not open in case of per-instance limit errors. Since now on the circuit breakers
+trigger the opening only if push and pull requests exceed the configured duration.
+
+The default inactivity timeout of active series in ingesters, controlled by the `-ingester.active-series-metrics-idle-timeout` configuration
+is increased from `10m` to `20m`.
+
+Query-frontend now returns the "413 Request Entity Too Large" error when a response shard from the `/active_series` request exceeds the limit.
```
Comment (56quarters) [inline docs/sources/mimir/release-notes/v2.14.md:59]: This doesn't really need to be called out.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+When executing a remote read request, the querier now honors the start/end time range, specified in the read hints.
+
+To prevent queue starvation with querier-worker queue prioritization algorithm, the querier now sets the minimum `-querier.max-concurrent` to four.
+Values below the allowed minimum are ignored.
+
+Distributor configuration's `-distributor.retry-after-header.max-backoff-exponent` and `-distributor.retry-after-header.base-seconds` are replaced
+with `-distributor.retry-after-header.min-backoff` and `-distributor.retry-after-header.max-backoff` for easier configuration.
+
+The circuit breakers in ingester are improved to not open in case of per-instance limit errors. Since now on the circuit breakers
```
Comment (56quarters) [inline docs/sources/mimir/release-notes/v2.14.md:53]: This is an implementation detail and probably doesn't need to be in release notes.

File: docs/sources/mimir/release-notes/v2.14.md (hunk 1 above)
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:84]:
```suggestion
There's an experimental Kafka-based ingest storage architecture that decouples the write and read paths through a Kafka-compatible backend.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+- the `-query-scheduler.use-multi-algorithm-query-queue` option in querier-scheduler
+- the YAML configuration `frontend.align_queries_with_step` in query-frontend
+
+## Experimental features
+
+Grafana Mimir 2.14 includes new features that are considered experimental and disabled by default.
+Use them with caution and report any issues you encounter:
+
+A new experimental Kafka-based ingest storage architecture is added, that decouples the write and read path through a Kafka-compatible backend.
+When enabled, distributors write incoming write requests to the Kafka-compatible backend, and the ingesters asynchronously replay ingested data from Kafka.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:85]:
```suggestion
When enabled, distributors write incoming write requests to the Kafka-compatible backend and the ingesters asynchronously replay ingested data from Kafka.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+New related configuration options that enable and configure the new architecture in Mimir are: `-ingest-storage.enabled`, `-ingest-storage.kafka.*`,
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:87]:
```suggestion
The following configuration options enable and configure the new architecture in Mimir: `-ingest-storage.enabled`, `-ingest-storage.kafka.*`,
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 14]
```diff
@@ -0,0 +1,139 @@
...
+`-ingest-storage.ingestion-partition-tenant-shard-size`, `-ingest-storage.read-consistency`, `-ingest-storage.migration.distributor-send-to-ingesters-enabled`,
+`-ingester.partition-ring.*`.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:89]: I think we could make this am unordered list instead of comma-seperated values for greater readability.

File: docs/sources/mimir/release-notes/v2.14.md (hunk 4 above)
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:91]:
```suggestion
The querier has an experimental streaming PromQL engine enabled by the`-querier.query-engine=mimir` configuration option.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 15]
```diff
@@ -0,0 +1,139 @@
...
+
+Querier has an experimental streaming PromQL engine. The configuration option `-querier.query-engine=mimir` enables the engine.
+
+Ingester added experimental `-ingester.ignore-ooo-exemplars` configuration. When set, the out of order exemplars are no longer reported
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:93]:
```suggestion
The ingester added an experimental `-ingester.ignore-ooo-exemplars` configuration. When set, out-of-order exemplars are no longer reported
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 16]
```diff
@@ -0,0 +1,139 @@
...
+to the remote write client.
+
+Querier now supports the experimental `limitk()` and `limit_ratio()` PromQL functions. The feature is disabled by default,
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:96]:
```suggestion
The querier supports the experimental `limitk()` and `limit_ratio()` PromQL functions. This feature is disabled by default,
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+but can be enabled with the `-querier.promql-experimental-functions-enabled=true` setting in the query-frontend and querier.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:97]:
```suggestion
but you can enable it with the `-querier.promql-experimental-functions-enabled=true` setting in the query-frontend and the querier.
```

File: docs/sources/mimir/release-notes/v2.14.md (hunk 5 above)
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:100]:
```suggestion
The header contains a comma-separated list of labels with values included in the logs.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+
+Querier supports for the experimental `X-Mimir-Chunk-Info-Logger` header, that triggers logging information about TSDB chunks loaded from ingesters and store-gateways in the querier.
+The header should contain the comma separated list of labels, for which their value will be included in the logs.
+
+Ruler includes the experimental `-ruler.rule-evaluation-write-enabled` configuration, that disables writing the result of rule evaluation to ingesters.
+This feature is meant to be used for testing.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:103]:
```suggestion
This feature is for testing purposes.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+
+## Bug fixes
+
+- Ruler: add support for draining any outstanding alert notifications before shutting down. This can be enabled with the `-ruler.drain-notification-queue-on-shutdown=true` CLI flag.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:107]:
```suggestion
- Ruler: add support for draining any outstanding alert notifications before shutting down.  Enable this setting with the `-ruler.drain-notification-queue-on-shutdown=true` CLI flag.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+- Store-gateway: fixed a case where, on a quick subsequent restart, the previous lazy-loaded index header snapshot was overwritten by a partially loaded one.
+- Ingester: fixed timestamp reported in the "the sample has been rejected because its timestamp is too old" error when the write request contains only histograms.
+- Store-gateway: store sparse index headers atomically to disk.
+- Query scheduler: fix a panic in request queueing.
+- Querier: fix issue where "context canceled" is logged for trace spans for requests to store-gateways that return no series when chunks streaming is enabled.
+- Alertmanager: Fix per-tenant silence limits not reloaded during runtime.
+- Alertmanager: Fixes a number of bugs in silences which could cause an existing silence to be deleted/expired when updating the silence failed. This could happen when the replacing silence was invalid or exceeded limits.
+- Alertmanager: Fix help message for utf-8-strict-mode.
+- Query-frontend: Ensure that internal errors result in an HTTP 500 response code instead of 422.
+- Configuration: Multi line envs variables are flatten during injection to be compatible with YAML syntax.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:121]:
```suggestion
- Configuration: Multi-line environment variables are flattened during injection to be compatible with YAML syntax.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+- Querier: fix issue where queries can return incorrect results if a single store-gateway returns overlapping chunks for a series.
+- HA Tracker: store correct timestamp for last received request from elected replica.
+- Querier: do not return `grpc: the client connection is closing` errors as HTTP `499`.
+- Compactor: fix a race condition between different compactor replicas that may cause a deleted block to be still referenced as non-deleted in the bucket index.
+- Querier: fix issue where some native histogram-related warnings were not emitted when `rate()` was used over native histograms.
+- Ruler: map invalid org-id errors to 400 status code.
+- Querier: Fix invalid query results when multiple chunks are being merged.
+- Query-frontend: return annotations generated during evaluation of sharded queries.
+- Querier: Support optional start and end times on `/prometheus/api/v1/labels`, `/prometheus/api/v1/label/<label>/values`, and `/prometheus/api/v1/series` when `max_query_into_future: 0`.
+- Alertmanager: Fix config validation gap around unreferenced templates.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:131]:
```suggestion
- Alertmanager: Fix configuration validation gap around unreferenced templates.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+- Alertmanager: Fix goroutine leak when stored config fails to apply and there is no existing tenant alertmanager.
```
Comment (tacole02) [inline docs/sources/mimir/release-notes/v2.14.md:132]:
```suggestion
- Alertmanager: Fix goroutine leak when stored configuration fails to apply and there is no existing tenant alertmanager.
```

File: docs/sources/mimir/release-notes/v2.14.md [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+
+The default inactivity timeout of active series in ingesters, controlled by the `-ingester.active-series-metrics-idle-timeout` configuration
+is increased from `10m` to `20m`.
+
+Query-frontend now returns the "413 Request Entity Too Large" error when a response shard from the `/active_series` request exceeds the limit.
+
+Following featues of store-gateway are changed: `-blocks-storage.bucket-store.max-concurrent-queue-timeout` is set to 5 seconds;
+`-blocks-storage.bucket-store.index-header.lazy-loading-concurrency-queue-timeout` is set to 5 seconds;
+`-blocks-storage.bucket-store.max-concurrent` is set to 200;
+
```
[the remark this answers is not in the document]
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md:64]: I'm not sure we should do it here. Nothing in the code related to redis configuration parameters are updated to be deprecated.

File: docs/sources/mimir/release-notes/v2.14.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md:53]: I've removed mentioning of this one, as suggested above.

File: docs/sources/mimir/release-notes/v2.14.md (hunk 14 above)
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md:89]: I've removed this one from the notes. The feature is still in the making.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 24]
```diff
@@ -0,0 +1,139 @@
...
+## Features and enhancements
+
+The minimal supported version of Go is updated to 1.22.
+
+The streaming of the chunks from store-gateways to queriers is enabled by default.
+
+Alertmanager adds the `-alertmanager.log-parsing-label-matchers` configuration option to control logging when parsing label matchers.
+This flag is intended to be used with `-alertmanager.utf8-strict-mode-enabled` to validate UTF-8 strict mode is working as intended.
+The default value for the configuration option is `false`.
```
[the remark this answers is not in the document]
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md]: I'm removing it from the notes.

File: docs/sources/mimir/release-notes/v2.14.md [hunk 25]
```diff
@@ -0,0 +1,139 @@
...
+
+Alertmanager adds the `-alertmanager.utf8-migration-logging-enabled` configuration option, that enables logging of tenant configurations,
+incompatible with UTF-8 strict mode. The default value for the configuration option is `false`.
```
[the remark this answers is not in the document]
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md]: Reading it now, I have no idea, why I decided to add it to the notes ✂️

File: docs/sources/mimir/release-notes/v2.14.md [hunk 26]
```diff
@@ -0,0 +1,139 @@
...
+Grafana Labs is excited to announce version 2.14 of Grafana Mimir.
+
+The highlights that follow include the top features, enhancements, and bug fixes in this release.
+For the complete list of changes, refer to the [CHANGELOG](https://github.com/grafana/mimir/blob/main/CHANGELOG.md).
+
+## Features and enhancements
+
+The minimal supported version of Go is updated to 1.22.
+
+The streaming of the chunks from store-gateways to queriers is enabled by default.
```
[the remark this answers is not in the document]
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md]: Added additional note about the feature

File: docs/sources/mimir/release-notes/v2.14.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (narqo) [reply docs/sources/mimir/release-notes/v2.14.md]: Added additional note about the feature.

**Comment by narqo** [comment]
🆙 Hey, all. Thank you for the feedback. I worked through all comments, and the PR is ready for another round of review

**Review by pstibrany — APPROVED** [review]
lgtm. (suggested some wording changes, but feel free to ignore)
